testing/e2e/e2e_withdrawal_test.go

```diff
@@ -0,0 +1,263 @@
+func (s *BeaconKitE2ESuite) getPendingPartialWithdrawals(stateID string) (*http.Response, error) {
+	client := s.initHTTPBeaconTest()
+
+	url := fmt.Sprintf("/eth/v1/beacon/states/%s/pending_partial_withdrawals", stateID)
+
+	resp, err := client.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get pending partial withdrawals: %w", err)
+	}
+	if resp == nil {
+		return nil, errors.New("received nil response")
+	}
+
+	return resp, nil
+}
+
+// checkPendingPartialWithdrawals checks if there are pending partial withdrawals
+// and returns the list of pending withdrawals if any
+func (s *BeaconKitE2ESuite) checkPendingPartialWithdrawals(stateID string) ([]types.PendingPartialWithdrawalData, error) {
+	resp, err := s.getPendingPartialWithdrawals(stateID)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var response types.PendingPartialWithdrawalsResponse
+	decoder := json.NewDecoder(resp.Body)
+	if err = decoder.Decode(&response); err != nil {
+		return nil, fmt.Errorf("failed to decode response: %w", err)
+	}
+
+	// Validate response fields
+	s.Require().Equal(response.Version, version.Name(version.Electra()))
+	s.Require().False(response.ExecutionOptimistic)
+	s.Require().True(response.Finalized)
+
+	withdrawals, ok := response.Data.([]types.PendingPartialWithdrawalData)
+	if !ok {
+		// Only if direct type assertion fails, fall back to JSON remarshal
+		dataBytes, errInMarshal := json.Marshal(response.Data)
+		if errInMarshal != nil {
+			return nil, fmt.Errorf("failed to marshal data: %w", errInMarshal)
+		}
+
+		if errInUnmarshal := json.Unmarshal(dataBytes, &withdrawals); errInUnmarshal != nil {
+			return nil, fmt.Errorf("failed to unmarshal withdrawals: %w", errInUnmarshal)
+		}
+	}
+
+	return withdrawals, nil
+}
+
+// findValidatorWithExecutionCredentials finds a validator with execution credentials
+func (s *BeaconKitE2ESuite) findValidatorWithExecutionCredentials(client *e2etypes.ConsensusClient) (string, crypto.BLSPubkey, error) {
+	// Get the validators to identify one with execution credentials (0x01)
+	validatorsResp, err := client.Validators(
+		s.Ctx(),
+		&beaconapi.ValidatorsOpts{
+			State: utils.StateIDHead,
+		},
+	)
+	if err != nil {
+		return "", crypto.BLSPubkey{}, fmt.Errorf("failed to get validators: %w", err)
+	}
+	if len(validatorsResp.Data) == 0 {
+		return "", crypto.BLSPubkey{}, errors.New("no validators found")
+	}
+
+	// Find a validator with execution withdrawal credentials (starting with 0x01)
+	var validatorIndex string
+	var blsPubkey crypto.BLSPubkey
+	for _, validator := range validatorsResp.Data {
+		credentials := validator.Validator.WithdrawalCredentials
+		if len(credentials) >= 2 && credentials[0] == 0x01 {
+			validatorIndex = fmt.Sprintf("%d", validator.Index)
+			// Convert the phase0.BLSPubKey to our crypto.BLSPubkey type
+			copy(blsPubkey[:], validator.Validator.PublicKey[:])
+			return validatorIndex, blsPubkey, nil
+		}
+	}
```

Comment:
it seems when users create validators through the deposit CLI, it creates withdrawal credentials with the 0x01 format by default (execution addresses). 
However I see test with non-ETH1 withdrawal credentials. Hence kept this check to ensure we get validator with execution withdrawal credentials for testing here.

Reply:
Fine, but is there a way to reuse the checks we have in production code instead of rewriting our own tests? Again in the interest of tests maintenability

Reply:
Can you pls elaborate, what checks are you referring to here ?

Reply:
Using method from consensus type  `IsValidEth1WithdrawalCredentials` for checking withdrawal creds. Let me know if you see any other checks that should be added.